exercises/practice/dnd-character/.meta/example.rb

```diff
@@ -0,0 +1,53 @@
+module Die
+  extend self
+
+  def roll(roll: 4, die: 6, remove_lowest: 1)
+    roll.times.map { rand(1..die) }.sort.pop(roll - remove_lowest).sum
+  end
+
+end
+
+module Modifiable
+
+  # This only calculates hit point modifier based on constitution
+  def modifier(constitution)
+    constitution/2 - 5
+  end
+
+end
+
+class DndCharacter
+  extend Modifiable
+
+  BASE_HITPOINTS = 10
+
+  private_constant :BASE_HITPOINTS
+
+  private
+
+  def initialize
+    @strength     = Die.roll
+    @dexterity    = Die.roll
+    @constitution = Die.roll
+    @intelligence = Die.roll
+    @wisdom       = Die.roll
+    @charisma     = Die.roll
+
+    @hitpoints    = BASE_HITPOINTS + self.class.modifier(constitution)
+  end
+
+  public
+
+  attr_reader :strength,
+              :dexterity,
+              :constitution,
+              :intelligence,
+              :wisdom,
+              :charisma,
+              :hitpoints
+
+end
+
+if $PROGRAM_NAME == __FILE__
+  puts DndCharacter.allocate.public_methods(false)
+end
```

Comment:
We can remove the library guard once we are done exploring.  Or at least make it something that is more "example" based rather than "debugging" like.

Reply:
Removed!